pkg/trace/api/api.go

```diff
@@ -516,13 +516,26 @@ func (r *HTTPReceiver) handleTraces(v Version, w http.ResponseWriter, req *http.
 		atomic.AddInt64(&ts.PayloadRefused, 1)
 		return
 	}
+	if err != nil {
+		e := err.Error()
+		if strings.Contains(e, "not found") {
+			log.Debugf("Failed to count traces: %s", err)
+		}
+		if strings.Contains(e, "not set") || strings.Contains(e, "not be parsed") {
+			log.Errorf("Failed to count traces: %s", err)
```

Comment:
I'm sorry but I don't think this is ok. If someone ever changes the error message, this won't work anymore. I'd propose to come up with an error constant, something like:
```go
var errInvalidTraceCount = fmt.Errorf("%q header value must be an integer.", headerTraceCount)
```
And return this error inside `traceCount` both when the value is empty and when it can not be parsed.

Then, you could just do this here:
```go
if err == errInvalidTraceCount {
    log.Error(err)
}
```

Reply:
Defined errors as variables not to use `strings.Contains()`. Also fix test codes as this change.
https://github.com/DataDog/datadog-agent/pull/11804/commits/eb92b71089eb779aff7c409eee82ab1590ee9572